models/comment_model.go

```diff
@@ -1,10 +1,23 @@
 package models
 
-import "google.golang.org/genproto/googleapis/type/date"
+import (
+	"time"
+
+	"cloud.google.com/go/firestore"
+)
 
 type Comment struct {
-	Comment_id int       `json:"comment_id"`
-	Content    string    `json:"content"`
-	Like       string    `json:"like,omitempty"`
-	Date       date.Date `json:"date"`
+	Content string                   `json:"content"`
+	Like    []*firestore.DocumentRef `json:"like,omitempty"`
+	Date    time.Time                `json:"date,omitempty"`
+}
+
```

Comment:
Actually, I may want to add 2 fields. 
that's is 
```
UserID   *firestore.DocumentRef   `json:"userId"`
PostID   *firestore.DocumentRef   `json:"postId"`
```

Reply:
ถ้าทำ collection เกี่ยวกับ service อะไร ไม่จำเป็น model ไม่จำเป็น นั้นเนื่องจาก 
มอง 1 model = 1 component  -> 1 model Comment = 1 Comment 
ซึ่งถ้ามองแบบนี้เวลาสร้าง หรือ create data ลง firebase เราจะให้มัน random id เอง 
ยกตัวอย่าง 
```
_, _, err := client.Collection("cities").Add(ctx, map[string]interface{}{
        "name":    "Tokyo",
        "country": "Japan",
})
if err != nil {
        // Handle any errors in an appropriate way, such as returning them.
        log.Printf("An error has occurred: %s", err)
}
```

อันนี้คือการเพิ่มข้อมูลลง collection cities จะเห็นว่าพอใช้คำสั่ง .Add แล้วมันไม่จำเป็นต้องใส่ id คำถามคือ document มันจะมีตัวอะไรระบุว่ามันมีอยู่ในระบบ ใช้อะไรเป็น id -> คำตอบคือ ถ้าใช้คำสั่ง .Add   document มันจะ random id ทำให้เราไม่จำเป็นจะต้องหา id มาใส่